### PR TITLE
Add missing fetch imports

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: packages/replay/src/auth.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import fetch from "node-fetch";
```
Comment (Andarist) [inline packages/replay/src/auth.ts:1]: This has been "broken"* for months from what I can tell but it seemed like a good idea to fix it now (I run an audit through all of the files in the repo to see if `fetch` is imported by all them). I decided to use `node-fetch` (over `undici`) as that's what is used more by this package (the old CLI)

*didn't always work in node 18 or earlier

File: packages/shared/src/cachedFetch.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { fetch } from "undici";
```
Comment (bvaughn) [inline packages/shared/src/cachedFetch.ts:1]: This seems _very easy_ to forget to do. Can we add a Lint rule to disallow using the global `fetch`? Else I can see us reintroducing this problem in the future.

Comment (Andarist) [reply packages/shared/src/cachedFetch.ts:1]: PRO-755 , it should be a super quick ticket but I'm wrapping up for today and I don't have time to switch branches and test it out. I'm also unsure what's the current state of lining in this repo

Comment (bvaughn) [reply packages/shared/src/cachedFetch.ts:1]: 🫡

**Pull request merged**